postprocessing/science/get_distances_to_rupture.jl

```diff
@@ -0,0 +1,195 @@
+# Script to calculate precise distances from SeisSol's surface/volume cells to the rupturing part of the fault
+# Output will be saved as a numpy file (.npy)
+# Multithreading in julia can be enabled with the flag -t 10
+# Example start command: mpiexecjl -np 5 julia -t 10 get_distances_to_rupture.jl data-surface.xdmf
+
+using SeisSolXDMF
+using ArgParse
+using NPZ
+
+function parse_commandline()
+    s = ArgParseSettings()
+    @add_arg_table s begin
+        "--slip_threshold"
+            help = "slip threshold to define the rupturing part of the fault"
+            arg_type = Float32
+            default = 0.1f0
+        "--use_gpu"
+            help = "enable GPU acceleration with CUDA.jl"
+            action = :store_true
+        "--event"
+            help = "usually 0; 1 and 2 events are needed for Ridgecrest where one simulation contains two events"
+            arg_type = Int
+            default = 0
+        "--noMPI"
+            help = "disable MPI; this flag will increase the performance of a single rank"
+            action = :store_true
+        "filename"
+            help = "path to SeisSol's surface or volume.xdmf"
+            required = true
+    end
+    return parse_args(s)
+end
+
+
+parsed_args = parse_commandline()
+filename = parsed_args["filename"]
+threshold = parsed_args["slip_threshold"]
+event = parsed_args["event"]
+use_gpu = parsed_args["use_gpu"]
+noMPI = parsed_args["noMPI"]
+
+
+function get_xyz(xdmf)
+    topology, geometry = grid_of(xdmf)
+    xyz = (1. / 3.) .* (geometry[:,topology[1,:]] .+ geometry[:,topology[2,:]] .+ geometry[:,topology[3,:]])
+    out = copy(transpose(convert(Matrix{Float32}, xyz)))
+    return out
+end
+
+
+function get_ASl(xdmf, event)
+    timesteps = timesteps_of(xdmf)
+    if event == 0
+        ASl = data_of(xdmf, size(timesteps)[1], "ASl")
+    elseif event == 1
+        ASl = data_of(xdmf, Int(round(size(timesteps)[1] / 2) - 1), "ASl")
+    elseif event == 2
+        ASl = data_of(xdmf, size(timesteps)[1], "ASl") .- data_of(xdmf, Int(round(size(timesteps)[1] / 2) - 1), "ASl")
+    else
+        println("Only 0, 1, 2 are allowed for the event argument.")
+        ccall(:jl_exit, Cvoid, (Int32,), 86) # Exit code 86 to the O/S
```

Comment:
Why are you using this very unidiomatic error handling?

Reply:
Changed to error()